src/nvim/api/vim.c

```diff
@@ -55,6 +56,64 @@ void nvim_command(String command, Error *err)
   try_end(err);
 }
 
+/// Retrieves highlight description from its name
+///
+/// @param name Highlight group name
+/// @return a highlight description e.g. {'bold': true, 'bg': 123, 'fg': 42}
+/// @see nvim_get_hl_by_id
+Dictionary nvim_get_hl_by_name(String name, Error *err)
+  FUNC_API_SINCE(3)
+{
+  Dictionary result = ARRAY_DICT_INIT;
+  int id = syn_name2id((const char_u *)name.data);
+
+  if (id == 0) {
+    api_set_error(err, kErrorTypeException, "Invalid highlight name %s",
+                  name.data);
+    return result;
+  }
+  result = nvim_get_hl_by_id(id, err);
+  return result;
+}
+
+/// Retrieves highlight description from its id (as returned by hlID())
+/// @see nvim_get_hl_by_name
+Dictionary nvim_get_hl_by_id(Integer hl_id, Error *err)
+  FUNC_API_SINCE(3)
+{
+  Dictionary dic = ARRAY_DICT_INIT;
+  if (syn_get_final_id((int)hl_id) == 0) {
+    api_set_error(err, kErrorTypeException, "Invalid highlight id %d", hl_id);
+    return dic;
+  }
+  int attrcode = syn_id2attr((int)hl_id);
+  return nvim_get_attr(attrcode, err);
+}
+
+Dictionary nvim_get_attr(Integer attr_id, Error *err)
```

Comment:
Needs a doc. Would nvim_get_text_attr_by_id be a better name? Or nvim_get_screen_cell_attr_by_id ?